File: c++/remove_duplicates.cpp
#include <vector>
#include <iostream>

static int RemoveDuplicates(std::vector<int>& vector)
{
	unsigned int c = 0;
	unsigned int k = vector.size();

	for(unsigned int i = 0; i < vector.size(); i++)
	{
		if(i != c && vector[i] == vector[c]) k--;

		if(vector[i] > vector[c])
		{
			c++;
			vector[c] = vector[i];
		}
	}

	return k;
}

int main()
{
	std::vector<int> vector = {1,1,1,1,2,2,2,2,3,3};

	unsigned int k = RemoveDuplicates(vector);

	for(unsigned int i = 0; i < k; i++)
	{
		std::cout << vector[i] << "\n";
	}
}

File: c#/RemoveDuplicates.cs
using System;

class Solution
{
	public static int RemoveDuplicates(int[] array)
	{
		int c = 0;
		int k = array.Length;

		for(int i = 0; i < array.Length; i++)
		{
			if(i != c && array[i] == array[c]) k--;

			if(array[i] > array[c])
			{
				c++;
				array[c] = array[i];
			}
		}

		return k;
	}
}

public class LeetCode
{
	public static void Main()
	{
		int[] array = {1,1,1,1,2,2,2,2,3,3};

		int k = Solution.RemoveDuplicates(array);

		for(int i = 0; i < k; i++)
		{
			Console.WriteLine("{0}", array[i]);
		}
	}
}
